lib/league_stat.rb

```diff
@@ -49,4 +51,22 @@ def count_of_teams
     stats_by_team.keys.size
   end
 
+  def best_offense
+    goals_per_game = {}
+    stats_by_team.each do |team|
+      total_goals = (team[1][:home_goals] + team[1][:away_goals]).to_f
+      goals_per_game[team[1][:team_name]] = (total_goals / team[1][:total_games]).round(2)
+    end
+    goals_per_game.max_by { |name, goals_per_game| goals_per_game }[0]
```

Comment:
this is the guts that will repeat, taking arguments for
* home, away, or total games
* goals scored or allowed
* highest or lowest score

all other structure will be identical for all eight methods. it will just pass three arguments to this method

Reply:
looks good, but method is getting kind of long, maybe remove logic in to helper methods